apps/src/code-studio/components/AssetThumbnail.jsx

```diff
@@ -46,39 +58,73 @@ class AssetThumbnail extends React.Component {
     style: PropTypes.object,
     iconStyle: PropTypes.object,
     useFilesApi: PropTypes.bool,
-    projectId: PropTypes.string
+    projectId: PropTypes.string,
+    src: PropTypes.string,
+    soundPlayer: PropTypes.object,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      isPlayingAudio: false
+    };
+    let api = this.props.useFilesApi ? filesApi : assetsApi;
+    if (this.props.projectId) {
+      api = api.withProjectId(this.props.projectId);
+    }
+    const basePath = api.basePath(this.props.name);
+    let cacheBustSuffix = '';
+    if (this.props.timestamp) {
+      const date = new Date(this.props.timestamp);
+      cacheBustSuffix = `?t=${date.valueOf()}`;
+    }
+    this.srcPath = `${basePath}${cacheBustSuffix}`;
+
+    if (this.props.type === 'audio') {
+      this.props.soundPlayer.register({id: this.srcPath, mp3: this.srcPath});
+    }
+  }
+
+  clickSoundControl = () => {
+    if (this.state.isPlayingAudio) {
+      this.setState({isPlayingAudio: false});
+      this.props.soundPlayer.stopPlayingURL(this.srcPath);
+    } else {
+      this.setState({isPlayingAudio: true});
+      this.props.soundPlayer.play(this.srcPath, {onEnded: ()=>{this.setState({isPlayingAudio: false});}});
+    }
   };
 
   render() {
     const {
-      timestamp,
       type,
-      name,
-      useFilesApi,
-      projectId,
       iconStyle,
-      style
+      style,
+      src
     } = this.props;
-    let api = useFilesApi ? filesApi : assetsApi;
-    if (projectId) {
-      api = api.withProjectId(projectId);
-    }
-    const basePath = api.basePath(name);
-    let cacheBustSuffix = '';
-    if (timestamp) {
-      const date = new Date(timestamp);
-      cacheBustSuffix = `?t=${date.valueOf()}`;
-    }
-    const srcPath = `${basePath}${cacheBustSuffix}`;
+
+    const playIcon = this.state.isPlayingAudio ? 'fa-pause-circle' : 'fa-play-circle';
 
     return (
-      <div className="assetThumbnail" style={[styles.wrapper, style]}>
-        {type === 'image' ?
-         <img src={srcPath} style={assetThumbnailStyle} /> :
-         <i
-           className={defaultIcons[type] || defaultIcons.unknown}
-           style={[assetIconStyle, iconStyle]}
-         />
+      <div>
+        {type === 'audio' ?
+          <div style={[styles.wrapper, styles.audioWrapper]}>
+            <i onClick={this.clickSoundControl} className={'fa '+ playIcon +' fa-4x'} style={styles.audioIcon} />
+          </div> :
+          <div className="assetThumbnail" style={[styles.wrapper, style, styles.background]}>
+            {type === 'image' ?
+              <a
+                href={src}
+                target="_blank"
+              >
+                <img src={this.srcPath} style={assetThumbnailStyle}/>
+              </a> :
+              <i
+                className={defaultIcons[type] || defaultIcons.unknown}
+                style={[assetIconStyle, iconStyle]}
+              />
+            }
+          </div>
```

Comment:
It might make sense here to extract some helper components, so you can write
```jsx
<div>
  {type === 'audio' ?
    <AudioThumbnail isPlaying={this.state.isPlayingAudio} onClick={this.clickSoundControl}/> :
    type === 'image' ?
    <ImageThumbnail src={this.srcPath} href={src}/> :
    <DefaultThumbnail type={type}/>}
</div>
```